packages/flutter/lib/src/material/search.dart

```diff
@@ -501,51 +514,47 @@ class _SearchPageState<T> extends State<_SearchPage<T>> {
       case null:
         break;
     }
-    final String routeName;
+    String? routeName;
```

Comment:
I am surprised that this was not already caught when we were updating to null safety, but a preferred way to handle this kind of case would be to have it be declared:

```dart
late finale routeName;
```

To prefer a non-nullable value.